internal/pkg/v2/dto/core/data/event.go

```diff
@@ -0,0 +1,49 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package data
+
+import (
+	dtoBase "github.com/edgexfoundry/edgex-go/internal/pkg/v2/dto/common/base"
+	model "github.com/edgexfoundry/edgex-go/internal/pkg/v2/model/core/data"
+)
+
+// AddEventRequest defines the Request Content for POST event DTO. This object and its properties correspond to the
+// AddEventRequest object in the APIv2 specification.
+type AddEventRequest struct {
+	dtoBase.Request `json:",inline"`
+	Device          string          `json:"device"`
+	Origin          string          `json:"origin,omitempty" codec:"origin,omitempty"`
+	Readings        []model.Reading `json:"readings,omitempty" codec:"readings,omitempty"`
+}
+
+// AddEventResponse defines the Response Content for POST event DTOs.  This object and its properties correspond to the
+// AddEventResponse object in the APIv2 specification.
+type AddEventResponse struct {
+	dtoBase.Response `json:",inline"`
+	ID               string `json:"id"` // ID uniquely identifies an event, for example a UUID
+}
+
+// EventCountResponse defines the Response Content for GET event count DTO.  This object and its properties correspond to the
+// EventCountResponse object in the APIv2 specification.
+type EventCountResponse struct {
+	dtoBase.Response `json:",inline"`
+	Count            int
+	DeviceID         string `json:"deviceId"` // ID uniquely identifies a device
+}
+
+// EventResponse defines the Response Content for GET event DTOs.  This object and its properties correspond to the
+// EventResponse object in the APIv2 specification.
+type EventResponse struct {
+	dtoBase.Response `json:",inline"`
+	Event            model.Event
+}
+
+// UpdateEventPushedRequest defines the Request Content for PUT event as pushed DTO. This object and its properties correspond to the
+// UpdateEventPushedByChecksumRequest object in the APIv2 specification.
+type UpdateEventPushedRequest struct {
```

Comment:
@tsconn23 in V2 is the Marked as Pushed only by checksum? From app services we currently only use checksum if eventId isn't set and checksum is set, i.e. checksum is set only when the event was CBOR encoded.

Reply:
@lenny-intel The DTO is defined in the spec here
https://github.com/edgexfoundry/edgex-go/blob/2cabbc24ce46fa1a5f5b1d576bede77ecf0c50f6/api/openapi/v2/core-data.yaml#L338

The naming of the above DTO called `UpdateEventPushedRequest` is incorrect according to the spec which the community has extensively reviewed and agreed to.

Reply:
@tsconn23 , Yes I expected to also see `UpdateEventPushedByIdRequest`.  Does Core Data create a checksum for all events or just CBOR encoded events? I.e. when we switch over to V2, we can drop the `UpdateEventPushedById` and just use `UpdateEventPushedByChecksum` for all events?

Reply:
> The naming of the above DTO called `UpdateEventPushedRequest` is incorrect.
DTO name update.

Reply:
Discussed Core WG 9-Apr-2020 -- Desirable to update by checksum only. Current implementation has a switch between ID (JSON) and Checksum (CBOR). @lenny-intel to investigate where the impl makes this determination and adds checksum to context.

https://github.com/edgexfoundry/go-mod-messaging/blob/cfba30792e745cabe7c9c124792dfed6ff0b4cfb/pkg/types/message_envelope.go#L44

Goal for V2 is to only update via the Checksum since it's common for all events regardless of originating content type.